docs/documentation/development/guidelines.md

```diff
@@ -26,13 +36,22 @@ Note that this list also serves as a checklist for code reviews on pull requests
 1. Packages that contain classes that are not meant to be used by other bundles should have "internal" in their package name.
 1. We are using [null annotations](https://wiki.eclipse.org/JDT_Core/Null_Analysis) from the Eclipse JDT project. Therefore every bundle should have an **optional** `Import-Package` dependency to `org.eclipse.jdt.annotation`.
 Classes should be annotated by `@NonNullByDefault` and return types, parameter types, generic types etc. are annotated with `@Nullable` only.
+For fields that store injected services, i.e. they will never be `null` in an OSGI framework, we annotate them with
```

Comment:
This is only valid for mandatory and static services - you might also directly want to mention OSGi Declarative Services here (which is named in the points above as well).